src/wallet/abstract-wallet.ts

```diff
@@ -2930,13 +2930,12 @@ abstract class AbstractWallet extends EventEmitter {
         for (let batchIndex = 0; batchIndex < totalBatches; batchIndex += 1) {
           const batchStart = startScanHeight + batchIndex * batchSize;
           const batchEnd = Math.min(batchStart + batchSize, treeHeight);
-          const fetcher = new Array<Promise<Optional<Commitment>>>(batchEnd);
 
-          for (let index = batchStart; index < batchEnd; index += 1) {
-            fetcher[index] = utxoMerkletree.getCommitment(treeIndex, index);
-          }
-
-          const leaves = await Promise.all(fetcher);
+          const leaves = await utxoMerkletree.getCommitmentRange(
+            treeIndex,
+            batchStart,
+            batchEnd - 1,
+          );
```

Comment:
I was just measuring `getCommitmentRange`. I don't understand how `scanLeaves` can be impacted by this, because `scanLeaves` doesn't call this code, and `scanLeaves` doesn't have its own batchSize either. Can you help me understand what kind of measurement you had done with `scanLeaves`?